components/director/internal/domain/integrationsystem/service.go

```diff
@@ -0,0 +1,93 @@
+package integrationsystem
+
+import (
+	"context"
+
+	"github.com/kyma-incubator/compass/components/director/internal/repo"
+
+	"github.com/pkg/errors"
+
+	"github.com/kyma-incubator/compass/components/director/internal/model"
+)
+
+//go:generate mockery -name=IntegrationSystemRepository -output=automock -outpkg=automock -case=underscore
+type IntegrationSystemRepository interface {
+	Create(ctx context.Context, item model.IntegrationSystem) error
+	Get(ctx context.Context, id string) (*model.IntegrationSystem, error)
+	List(ctx context.Context, pageSize int, cursor string) (model.IntegrationSystemPage, error)
+	Update(ctx context.Context, model model.IntegrationSystem) error
+	Delete(ctx context.Context, id string) error
+}
+
+//go:generate mockery -name=UIDService -output=automock -outpkg=automock -case=underscore
+type UIDService interface {
+	Generate() string
+}
+
+type service struct {
+	intSysRepo IntegrationSystemRepository
+
+	uidService UIDService
+}
+
+func NewService(intSysRepo IntegrationSystemRepository, uidService UIDService) *service {
+	return &service{
+		intSysRepo: intSysRepo,
+		uidService: uidService,
+	}
+}
+
+func (s *service) Create(ctx context.Context, in model.IntegrationSystemInput) (string, error) {
+	id := s.uidService.Generate()
+	intSys := in.ToIntegrationSystem(id)
+
+	err := s.intSysRepo.Create(ctx, intSys)
+	if err != nil {
+		if repo.IsNotUnique(err) {
+			return "", errors.New("Integration System name is not unique")
+		}
+		return "", err
+	}
+
+	return id, nil
+}
+
+func (s *service) Get(ctx context.Context, id string) (*model.IntegrationSystem, error) {
```

Comment:
Please also add the following method: `Exist(ctx context.Context, id string) (bool, error)`